### PR TITLE
Lexicon: Add support for subtraction

### LINKED ISSUE #273: Subtraction isn't implemented
```coffee
a = 1 - 2
```

yields:

<details>

```json
{
  "tree": {
    "1": {
      "method": "expression",
      "ln": "1",
      "name": [
        "a"
      ],
      "args": [
        {
          "$OBJECT": "expression",
          "expression": "subtraction",
          "values": [
            {
              "$OBJECT": "int",
              "int": 1
            },
            {
              "$OBJECT": "int",
              "int": 2
            }
          ]
        }
      ],
      "src": "a = 1 - 2"
    }
  },
  "entrypoint": "1",
  "version": "0.17.3-dirty"
}
```

</details>

### PR DESCRIPTION
**Pull request by assimovt**
Fixes #273

### CODE REVIEW

File: asyncy/utils/Resolver.py [hunk 1]
```diff
@@ -168,6 +169,11 @@ def expression(cls, item, data):
                 result += r
 
             return result
+        elif a == 'subtraction':
+            right = cls.resolve(values[1], data)
+            assert type(left) in (int, float)
```
Comment (wilzbach) [inline asyncy/utils/Resolver.py]: Storyscript supports implicit casts of boolean to int (like e.g. Python does too). We never decided whether we want the compiler to insert casts or the engine to support it, but in the latter case the at assert would need to be expanded.

Comment (assimovt) [reply asyncy/utils/Resolver.py]: @wilzbach gotcha, thanks for clarifying. shall we open a separate issue to support that for other ops as well?

Comment (wilzbach) [reply asyncy/utils/Resolver.py]: Well, we first need to decide whether the engine or the SS compiler is supposed to do the casting, but I guess as long as we have any, the following is possible:

```coffee
a = {}  # Map[any,any]
a[0] = true
b = a[0] # any
c = b + 1 # 2
```

**Comment by judepereira** [comment]
@wilzbach what's the status of this PR? Can we merge it?

**Pull request merged**